File: common/src/main/java/example/common/entity/Department.java
package example.common.entity;

import java.util.ArrayList;
import java.util.List;
import org.seasar.doma.Association;
import org.seasar.doma.Entity;
import org.seasar.doma.Id;
import org.seasar.doma.Metamodel;
import org.seasar.doma.Version;

@Entity(metamodel = @Metamodel)
public class Department {
  @Id Integer id;

  String name;

  @Version Integer version;

  @Association List<Employee> employees = new ArrayList<>();

  public Integer getId() {
    return id;
  }

  public void setId(Integer id) {
    this.id = id;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public Integer getVersion() {
    return version;
  }

  public void setVersion(Integer version) {
    this.version = version;
  }

  public List<Employee> getEmployees() {
    return employees;
  }
}

File: dsl-style-kotlin/src/main/kotlin/example/dsl_style_kotlin/entity/Department.kt
package example.dsl_style_kotlin.entity

import org.seasar.doma.Entity
import org.seasar.doma.Id
import org.seasar.doma.Metamodel
import org.seasar.doma.Transient
import org.seasar.doma.Version

@Entity(metamodel = Metamodel())
class Department {
    @Id
    var id: Int? = null
    var name: String? = null
    @Version
    var version: Int? = null
    @Transient
    val employees: MutableList<Employee> = mutableListOf()
}
